python/pyspark/pandas/data_type_ops/date_ops.py

```diff
@@ -80,35 +76,33 @@ def rsub(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         )
         if isinstance(right, datetime.date) and not isinstance(right, datetime.datetime):
             warnings.warn(msg, UserWarning)
-            return -column_op(cast(Callable[..., GenericColumn], F.datediff))(
-                left, F.lit(right)
-            ).astype("long")
+            return -column_op(F.datediff)(left, F.lit(right)).astype("long")
         else:
             raise TypeError("Date subtraction can only be applied to date series.")
 
     def lt(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         from pyspark.pandas.base import column_op
 
         _sanitize_list_like(right)
-        return column_op(cast(Callable[..., GenericColumn], Column.__lt__))(left, right)
+        return column_op(Column.__lt__)(left, right)
```

Comment:
This seems like always using non Spark Column. Is this correct?

Reply:
Seems like we should branch `ConnectColumn` and `PySparkColumn`. Let me address it